src/components/middle/immedia/Immedia.tsx

```diff
@@ -229,164 +264,124 @@ const Immedia: FC<OwnProps & StateProps> = ({ chatId, currentUser }) => {
+          // FIX: Maybe there's a better way to do this.
+          // TRY: https://developer.mozilla.org/en-US/docs/Web/API/ImageCapture/ImageCapture
+          setTimeout(() => {
+            const video = videoMeRef.current;
+            const canvas = canvasMeRef.current;
+            // eslint-disable-next-line  no-null/no-null
+            if (video === null || canvas === null) return;
+            // show snapshot
+            context.drawImage(
+              video,
+              160,
+              120,
+              360,
+              240,
+              0,
+              0,
+              canvas.width,
+              canvas.height,
+            );
+            const image = canvas.toDataURL('image/png');
+            setLastSnapshot(image);
+          }, SNAPSHOT_RATE / 5);
         }
-      }
-    };
+      };
 
-    let result: NodeJS.Timeout;
-    if (awareness) {
-      // get webcam snapshots every SNAPSHOT_RATE seconds
-      result = setInterval(getSnapshotVideo, SNAPSHOT_RATE);
+      if (navigator.mediaDevices.getUserMedia) {
+        // TODO: Rewrite using async/await
+        navigator.mediaDevices
+          .getUserMedia({ video: true, audio: false })
+          .then((stream) => cbk(stream))
+          .catch((err) => {
+            // eslint-disable-next-line no-console
+            console.error(err);
+          });
+      } else {
+        // eslint-disable-next-line no-console
+        console.error(new Error(`${INIT}There is no user media`));
+      }
     }
-    return () => clearInterval(result);
-  }, [awareness]);
+  };
+
+  useInterval(() => {
+    getSnapshotVideo();
+  }, awareness ? SNAPSHOT_RATE : undefined);
 
   // GC that runs every GC_RATE seconds and checks if, for each participant,
   // their last snapshot was taken inside a REMOVE_THRESHOLD seconds time frame.
   // If not, it will remove the participant.
-  useEffect(() => {
-    const updateParticipants = () => {
-      console.log(INIT, 'Garbage collect participants');
-      participants.forEach((participant) => {
-        const removeThreshold = new Date().getTime() - REMOVE_THRESHOLD;
-        if (participant.timestamp && participant.timestamp < removeThreshold) {
-          console.log(INIT, 'Garbage collect participant', participant.id);
-          setParticipants(participants.filter((p) => p.id !== participant.id));
-        }
-      });
-      if (participants.length === 0) {
-        console.log(INIT, 'There is 1 participant left');
-      } else {
-        console.log(
-          INIT,
-          'There are',
-          1 + participants.length,
-          'participants left.',
-        );
+  const updateParticipants = () => {
+    participants.forEach((participant) => {
+      const removeThreshold = new Date().getTime() - REMOVE_THRESHOLD;
+      if (participant.timestamp && participant.timestamp < removeThreshold) {
+        setParticipants(participants.filter((p) => p.id !== participant.id));
```

Comment:
I think it would be nice to log the ammount of participants Immedia currently has.

Reply:
Done in 5c13275
I prefered to add the log in a new `useEffect` because it was cleaner.